standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5059,6 +5065,11 @@ private boolean drop_partition_common(RawStore ms, String catName, String db_nam
       tableDataShouldBeDeleted = checkTableDataShouldBeDeleted(tbl, deleteData);
       firePreEvent(new PreDropPartitionEvent(tbl, part, deleteData, this));
       mustPurge = isMustPurge(envContext, tbl);
+      try {
+        writeId = getWriteId(envContext);
+        writeTruncatedBase = !deleteData && writeId > 0;
+      } catch (NumberFormatException e) {
+      }
```

Comment:
nit: Should we at least log the exception?

Reply:
refactored